src/odb/src/db/dbWireCodec.cpp

```diff
@@ -642,6 +642,32 @@ void dbWireEncoder::clearColor()
   addOp(WOP_COLOR, 0);
 }
 
+void dbWireEncoder::setViaColor(uint8_t bottom_color,
+                                uint8_t cut_color,
+                                uint8_t top_color)
+{
+  // LEF/DEF says 3 is the max number of supported masks per layer.
+  // 0 is also not a valid mask.
+  for (const auto color : {bottom_color, cut_color, top_color}) {
+    if (color > 3) {
+      utl::Logger* logger = _wire->getImpl()->getLogger();
+      logger->error(
+          utl::ODB, 1103, "Mask color: {}, but must be between 0 and 3", color);
+    }
+  }
+
+  // encode as XX BB CC TT
+  const uint8_t mask_color = bottom_color << 4 | cut_color << 2 | top_color;
```

Comment:
You should just make a struct with bit fields, and then cast it to uint8 in the WOP. It'll save you from having to do the bit twiddling manually

Reply:
I would be all for that, but I haven't tried that before and a quick stab at it didn't work, so I'll probably need some help as to how to do that.

Reply:
@QuantamHD I'm inclined to merge and you can PR your idea as an improvement.